cpp/src/parquet/encoding_test.cc

```diff
@@ -635,6 +635,146 @@ TEST(BooleanArrayEncoding, AdHocRoundTrip) {
   }
 }
 
+class TestBooleanArrowDecoding : public ::testing::Test {
+ public:
+  void SetUp() override {
+    null_probabilities_ = {0.0, 0.001, 0.5, 0.999, 1.0};
+    read_batch_sizes_ = {1024, 4096, 10000};
+    true_probabilities_ = {0.0, 0.001, 0.5, 0.999, 1.0};
```

Comment:
Only true_prob = 0.999 but not 1 trigger the bug in issue, because if true_prob = 1, it goes to fastpath without hittin a all 1 block